infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuInterClusterReplicationServerNode.java

```diff
@@ -1,82 +1,70 @@
 package org.corfudb.infrastructure.logreplication.infrastructure;
 
-import com.google.common.collect.ImmutableMap;
 import com.google.common.util.concurrent.ThreadFactoryBuilder;
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
-import org.corfudb.infrastructure.AbstractServer;
-import org.corfudb.infrastructure.BaseServer;
 import org.corfudb.infrastructure.ServerContext;
-import org.corfudb.infrastructure.ServerThreadFactory;
-import org.corfudb.infrastructure.logreplication.runtime.LogReplicationServerRouter;
+import org.corfudb.infrastructure.logreplication.runtime.LogReplicationClientServerRouter;
 
 import javax.annotation.Nonnull;
-import java.util.ArrayList;
-import java.util.Map;
-import java.util.concurrent.CompletableFuture;
-import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.atomic.AtomicBoolean;
 
+/**
+ * This class manages the lifecycle of the transport layer server
+ */
 @Slf4j
 public class CorfuInterClusterReplicationServerNode implements AutoCloseable {
 
     @Getter
     private final ServerContext serverContext;
 
-    @Getter
-    private final Map<Class, AbstractServer> serverMap;
-
-    @Getter
-    private final LogReplicationServerRouter router;
-
-    // This flag makes the closing of the CorfuServer idempotent.
-    private final AtomicBoolean close;
-
-    private LogReplicationServer logReplicationServer;
-
     private ScheduledExecutorService logReplicationServerRunner;
 
     // Error code required to detect an ungraceful shutdown.
     private static final int EXIT_ERROR_CODE = 100;
 
+    // this flag makes the start/stop operation of transportLayerServer idempotent
+    @Getter
+    private AtomicBoolean serverStarted;
```

Comment:
Just wondering since this variable is also used below via close and cleanUp, should this be something like `runLRServer` ?

Reply:
Are you suggesting we rename it or slightly change what this variable indicates?
So then "runLRServer" would be true when the server is not started, indicating we can trigger the start of server..and it would be false when the server is already running..

for rename it, would LrServerRunning be better?

Reply:
Naming is hard - sure something like `isLrServerRunning` perhaps?

Reply:
Sorry thinking again I think startServer is fine